conformance/grammar.isl

```diff
@@ -251,6 +251,15 @@ type::{
   ]
 }
 
+type::{
+  name: model_float,
+  type: string,
+  // optional '-', one or more digits, '.', zero or more digits, optional group of 'e' and one or more digits OR
+  //     optional '-', one or more digits, 'e', one or more digits OR
+  //     one of the non numeric float values
+  regex: "^-?\\d+[.]\\d*(e\\d+)?|-?\\d+e\\d+|[+-]inf|nan$"
```

Comment:
I wondered about this earlier, but why allow the conformance tests to express floats using the same syntax as Ion decimals? Why is the `e` optional?

Reply:
There was a bit of a TBD here, I guess, that didn't really get resolved. TLDR; `model-float` relies on the IEEE-754 string representation of floating point numbers.

We had a bit of a bootstrapping problem when it comes to testing floats. Unless we want to have `model-float` define values in terms of a sign, coefficient, and mantissa, we need to have some alternate representation. I took the lazy/simple way and said that it should be a string that can be parsed by a floating point library (usually in the standard library of a language). Then, I ran into the problem that these libraries don't always agree on the serialization of non-numeric/finite values, so I had to specify that in a way that excluded things like `+Infinity`, `+Inf`, etc. The result is that I had to specify something for the finite float values. I could have done `[0-9.e-]+`, but that seemed too sloppy, so I approximated what is allowed in IEEE-754.

Anyway, although it might make it look like there's a bit of a circular definition, I don't mind changing it to Ion float syntax.